services/comments.js

```diff
@@ -44,29 +45,87 @@ module.exports = class CommentsService {
 
   /**
    * Adds a tag if it doesn't already exist on the comment.
+   * @throws if tag is already added to the comment
+   * @throws if tag name is not in ALLOWED_TAGS
+   * @param {String} id the id of the comment to tag
+   * @param {String} name the name of the tag to add
+   * @param {String} assigned_by the user id for the user who added the tag
    */
   static addTag(id, name, assigned_by) {
 
-    if (ALLOWED_TAGS.find((t) => t.name === name) == null) {
+    if (ALLOWED_TAGS.find((t) => t === name) == null) {
       return Promise.reject(new Error('tag not allowed'));
     }
 
-    return CommentModel.update({
+    const filter = {
       id,
-      tags: {
-        $ne: {
-          name
+      tags: {$not: {$elemMatch: {name}}}
```

Comment:
Should just use the previous query here, since `$elemMatch` provides only a single query, it is not necessary, see https://docs.mongodb.com/manual/reference/operator/query/elemMatch/#single-query-condition